Modules/hpe3par_snapshot.py

```diff
@@ -435,13 +463,190 @@ def restore_snapshot_online(
+               else:
+                  if (len(day_of_week_task)>1 or day_of_week_task == ""):
+                    return (False, False, "Invalid task frequency day of week", {})
+           else:
+              return (False, False, "Invalid task frequency string", {})
+       else:
+             return (False, False, "Invalid task frequency string", {})
+       freq = task_freq_custom        
+    try:
+        client_obj.login(storage_system_username, storage_system_password)
+        client_obj.setSSHOptions(storage_system_ip, storage_system_username, storage_system_password)
+        if not client_obj.volumeExists(base_volume_name):
+           return (False, False, "Volume does not Exist", {})
+
+        if not client_obj.scheduleExists(schedule_name):
+           cmd = ["createsv"]
+           if read_only:
+              cmd.append("-ro")
+           if expiration_time:
+             cmd.append("-exp")
+             cmd.append(str(expirationHours)+"h")
+           if retention_time:
+             cmd.append("-retain")
+             cmd.append(str(retentionHours)+"h")
+           snap_string = ".@y@@m@@d@@H@@M@@S@"
+           cmd.append(snapshot_name+snap_string)
+           cmd.append(base_volume_name)
+           if task_freq:
+              freq="@"+task_freq
+           cmd = ' '.join(cmd)
+           client_obj.createSchedule(
+                schedule_name, cmd, freq)
+        else:
+            return (True, False, "Schedule already Exist", {})
+    except Exception as e:
+        return (False, "False", "Schedule creation failed | %s" % (e), {})
+    finally:
+        client_obj.logout()
+    return (
+        True,
+        True,
+        "Created Schedule %s successfully." %
+       schedule_name,
+        {})
+                
+def delete_schedule(
+        client_obj,
+        storage_system_ip,
+        storage_system_username,
+        storage_system_password,
+        schedule_name):
+    if storage_system_username is None or storage_system_password is None:
+        return (
+            False,
+            False,
+            "Schedule delete failed. Storage system username or password is \
+null",
+            {})
+    if schedule_name is None:
+        return (
+            False,
+            False,
+            "Schedule delete failed. Schedule name is null",
+            {})
+    if len(schedule_name) < 1 or len(schedule_name) > 31:
+        return (False, False, "Schedule create failed. Schedule name must be atleast 1 character and not more than 31 characters", {})
+    try:
+        client_obj.setSSHOptions(storage_system_ip, storage_system_username, storage_system_password)
+        if client_obj.scheduleExists(schedule_name):
+           client_obj.deleteSchedule(schedule_name)
+        else:
+           return (True, False, "Schedule does not exist", {})
+    except Exception as e:
+        return (False, False, "Schedule delete failed | %s" % (e), {})
+    finally:
```

Comment:
This is not correct. Please remove finally instead of passing inside as it makes no sense

Reply:
ok